Bug 1400754 - stylo: crash on Win64 Asan build.  r=manishearth, dmajor.

julian-seward1 opened the pull request:
* adds a hashglobe::alloc::realloc, as that was previously not implemented,
  copying and simplifying from liballoc_system.

* routes malloc and realloc calls through hashglobe::alloc::, instead of
  doing it via direct 'extern "C"' calls.

* removes the #[cfg(feature = "known_system_malloc")] markings in
  fallible/lib.rs (the fallible vec stuff) on the assumption that we no
  longer need to special-case these uses.

<!-- Please describe your changes on the following line: -->


---
<!-- Thank you for contributing to Servo! Please replace each `[ ]` by `[X]` when the step is complete, and replace `__` with appropriate data: -->
- [ ] `./mach build -d` does not report any errors
- [ ] `./mach test-tidy` does not report any errors
- [ ] These changes fix #__ (github issue number if applicable).

<!-- Either: -->
- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

<!-- Also, please make sure that "Allow edits from maintainers" checkbox is checked, so that we can help you if you get stuck somewhere along the way.-->

<!-- Pull requests that do not address these steps are welcome, but they will require additional verification as part of the review process. -->

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/servo/servo/18661)
<!-- Reviewable:end -->

highfive commented:
<img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20> **Warning** <img src="http://www.joshmatthews.net/warning.svg" alt="warning" height=20>

* These commits modify **unsafe code**. Please review it carefully!

jdm commented:
@bors-servo: delegate+

bors-servo commented:
:v: @julian-seward1 can now approve this pull request

emilio commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 9a00538384dd96192862fb984fd894f4b3d4f9d2 with merge d0344f998d93f004499f985bef520bacee5d9925...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/9463)

julian-seward1 commented:
@bors-servo: retry

bors-servo commented:
:hourglass: Trying commit 9a00538384dd96192862fb984fd894f4b3d4f9d2 with merge 14aa8895d75dcb69e28695740d6bf84c3a006341...

bors-servo commented:
:broken_heart: Test failed - [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/9464)

julian-seward1 committed:
Bug 1400754 - stylo: crash on Win64 Asan build.  r=manishearth, dmajor.

* adds a hashglobe::alloc::realloc, as that was previously not implemented,
  copying and simplifying from liballoc_system.

* routes malloc and realloc calls through hashglobe::alloc::, instead of
  doing it via direct 'extern "C"' calls.

mbrubeck commented:
@bors-servo try

bors-servo commented:
:hourglass: Trying commit 419642a2bd7b15ad8b6e8bbe68a8dd6f9e778174 with merge d9121598974b71d5a1f10038df7e3cffefa942dc...

bors-servo commented:
:broken_heart: Test failed - [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/6314)

julian-seward1 commented:
@bors-servo: retry

bors-servo commented:
:hourglass: Trying commit 419642a2bd7b15ad8b6e8bbe68a8dd6f9e778174 with merge f90db81c94aad46a3f739cd598e089d5527d0e25...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/9145), [arm32](http://build.servo.org/builders/arm32/builds/9380), [arm64](http://build.servo.org/builders/arm64/builds/9420), [linux-dev](http://build.servo.org/builders/linux-dev/builds/9458), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/6266), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/6318), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/9516), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/1960), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/1967), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/6214), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/6275), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/1997), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/1990), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/4780)
State: approved= try=True

emilio commented:
@bors-servo r+

bors-servo commented:
:pushpin: Commit 419642a has been approved by `emilio`

<!-- @bors-servo r=emilio 419642a2bd7b15ad8b6e8bbe68a8dd6f9e778174 -->

bors-servo commented:
:hourglass: Testing commit 419642a2bd7b15ad8b6e8bbe68a8dd6f9e778174 with merge 04de056cde3c2626e2c8a4910dd69c9b5f9a5e11...

bors-servo commented:
:sunny: Test successful - [android](http://build.servo.org/builders/android/builds/9148), [arm32](http://build.servo.org/builders/arm32/builds/9383), [arm64](http://build.servo.org/builders/arm64/builds/9422), [linux-dev](http://build.servo.org/builders/linux-dev/builds/9460), [linux-rel-css](http://build.servo.org/builders/linux-rel-css/builds/6268), [linux-rel-wpt](http://build.servo.org/builders/linux-rel-wpt/builds/6320), [mac-dev-unit](http://build.servo.org/builders/mac-dev-unit/builds/9519), [mac-rel-css1](http://build.servo.org/builders/mac-rel-css1/builds/1962), [mac-rel-css2](http://build.servo.org/builders/mac-rel-css2/builds/1969), [mac-rel-wpt1](http://build.servo.org/builders/mac-rel-wpt1/builds/6217), [mac-rel-wpt2](http://build.servo.org/builders/mac-rel-wpt2/builds/6278), [mac-rel-wpt3](http://build.servo.org/builders/mac-rel-wpt3/builds/1999), [mac-rel-wpt4](http://build.servo.org/builders/mac-rel-wpt4/builds/1992), [windows-msvc-dev](http://build.servo.org/builders/windows-msvc-dev/builds/4783)
Approved by: emilio
Pushing 04de056cde3c2626e2c8a4910dd69c9b5f9a5e11 to master...
